src/main/java/net/minecraftforge/event/level/AlterGroundEvent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.level;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.LevelSimulatedReader;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecorator;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * This event is fired when {@link net.minecraft.world.level.levelgen.feature.treedecorators.AlterGroundDecorator#placeBlockAt(TreeDecorator.Context, BlockPos)}
+ * attempts to alter a ground block when generating a feature. An example of this would be large spruce trees converting grass blocks into podzol.
+ * <p>
+ * This event is {@linkplain Cancelable cancellable}.
+ * <p>
+ * This event is fired on the {@linkplain MinecraftForge#EVENT_BUS main Forge event bus}
+ * only on the {@linkplain net.minecraftforge.fml.LogicalSide#SERVER logical server}.
+ */
+@Cancelable
+public class AlterGroundEvent extends Event {
+    private final LevelSimulatedReader level;
+    private final RandomSource random;
+    private final BlockPos pos;
+    private BlockState altered;
+
+    public AlterGroundEvent(LevelSimulatedReader level, RandomSource random, BlockPos pos, BlockState altered) {
+        super();
+        this.level = level;
+        this.random = random;
+        this.pos = pos;
+        this.altered = altered;
+    }
+
+    public LevelSimulatedReader getLevel() {
+        return this.level;
+    }
+
+    public RandomSource getRandom() {
+        return this.random;
+    }
+
+    /**
+     * @return The position of the block that will be altered.
+     */
+    public BlockPos getPos() {
+        return this.pos;
+    }
+
+    /**
+     * @return The new block state to be placed by the ground decorator.
+     */
+    public BlockState getAlteredState() {
+        return this.altered;
+    }
```

Comment:
I think there should probably be two separate fields: one for the original altering state and one for the altered state. Just more safety as some mods would probably attempt to alter the same block and if the state changes for one then they would have no idea what the original was.